operator/src/main/resources/scripts/startServer.sh

```diff
@@ -290,12 +290,27 @@ createFolder ${DOMAIN_HOME}/servers/${SERVER_NAME}/security
 copyIfChanged /weblogic-operator/introspector/boot.properties \
               ${DOMAIN_HOME}/servers/${SERVER_NAME}/security/boot.properties
 
+# remove write and execute permissions for group to prevent insecure file system warnings.
+chmod g-wx  ${DOMAIN_HOME}/servers/${SERVER_NAME}/security/boot.properties
+
+
 if [ ${DOMAIN_SOURCE_TYPE} != "FromModel" ]; then
   trace "Copying situational configuration files from operator cm to ${DOMAIN_HOME}/optconfig directory"
   copySitCfgWhileBooting /weblogic-operator/introspector ${DOMAIN_HOME}/optconfig             'Sit-Cfg-CFG--'
   copySitCfgWhileBooting /weblogic-operator/introspector ${DOMAIN_HOME}/optconfig/jms         'Sit-Cfg-JMS--'
   copySitCfgWhileBooting /weblogic-operator/introspector ${DOMAIN_HOME}/optconfig/jdbc        'Sit-Cfg-JDBC--'
   copySitCfgWhileBooting /weblogic-operator/introspector ${DOMAIN_HOME}/optconfig/diagnostics 'Sit-Cfg-WLDF--'
+else
+  if [[ ! -z ${CLOUD_PLATFORM} && ${CLOUD_PLATFORM} == "Openshift" ]]; then
+    # Operator running on Openshift platform - change file permissions in the DOMAIN_HOME dir to give
+    # group same permissions as user .
+    chmod -R g=u ${DOMAIN_HOME} || return 1
```

Comment:
Should we also do it for non mii case? Or this is left as an exercise for the user when they create the domain?

Reply:
@jshum2479 

I originally suggested that we handle the non-MII cases, but I have a similar concern.  In detail:

For one, a recursive file traversal of the entire domain-home seems like it could be prohibitively expensive, especially as it's done for every pod cycle. 

For DII in particular, maybe this can this occur solely when the image is created (and be left up to the customer)? 

And for DiPV, if we must do something automatic, can it be limited to occur only in the introspector, and, even then, only the first time the introspector sees the domain?

Reply:
I agreed, doing it in the introspector (before zipping up the domain) is a better place.  Do we really need -R or is the check only for bin directory?

Reply:
@rjeberhard FYI

Reply:
Also, have we explored doing it in WDT?

Reply:
Sorry, one more question.  Does the new was check provide any guidance on JKS keystore files?   Best practice is user read only.

Reply:
I'll set up a meeting to go over this. As per the requirements in JIRA OWLS-91448, we should NOT use `chmod -R g=u` for Model-in-image domain unless the environment is OpenShift. My understanding from JIRA was that, DII case will be handled in image tool and it's customers responsibility to do it for domain-on-pv.

Reply:
I have made the changes to perform the `chmod -R g=u ${DOMAIN_HOME}` in the introspector for model-in-image domain and to perform `chmod -R g=u ${DOMAIN_HOME}` in startServer.sh script for domain-home-in-image. My understanding is that we should not be doing `chmod -R` for the domain-on-pv domains since those changes will be persistent.

Reply:
If the image tool is going to handle it  then why does startServer need to? (You wrote _"DII case will be handled in image tool "_ and _"I have made the changes to perform ...  chmod -R g=u ${DOMAIN_HOME} in startServer.sh script for domain-home-in-image."_)

Reply:
My comment "DII case will be handled in the image tool" on 9/13 was based on my previous understanding. However, in the standup meeting discussion on Tuesday 9/14, Derek and Robert Patrick suggested to also do `chmod -R u=g ${DOMAIN_HOME}` for DII in the operator.